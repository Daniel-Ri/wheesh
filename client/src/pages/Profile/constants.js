export const SET_PROFILE = 'Profile/SET_PROFILE';
export const GET_PROFILE = 'Profile/GET_PORFILE';
export const UPDATE_PROFILE = 'Profile/UPDATE_PROFILE';
