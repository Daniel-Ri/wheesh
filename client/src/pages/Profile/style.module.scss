@import '@styles/base';

// Reset
* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  font-family: Poppins;
}

.main {
  margin-left: 200px;
  position: relative;
  background: $color-alabaster;
  display: flex;
  justify-content: center;
  width: calc(100vw - 200px);
  height: 100%;
  min-height: 100vh;

  .container {
    width: 90%;
    max-width: 40rem;
    @include directionY;

    header {
      width: 100%;
      margin-top: 5vh;
      @include directionX;
      justify-content: space-between;
      align-items: center;
      position: relative;

      .empty {
        flex: 32%;
      }

      h1 {
        font-size: 120%;
      }

      .edit {
        flex: 32%;
        width: fit-content;
        text-align: right;
        cursor: pointer;
        text-transform: capitalize;
      }
    }
    
    form {
      margin-top: 1rem;
      width: 100%;
      @include directionY;
  
      .header {
        padding: 0.5rem;
        background-color: $color-light-gray;
      }
  
      label {
        flex: 45%;
      }
  
      div.input {
        width: 100%;
        padding: 0.5rem;
        @include directionX;
        justify-content: space-between;
        align-items: center;
      
        input {
          flex: 45%;
          width: 45%;
          padding: 0.25rem 0.5rem;
          font-size: 80%;
          text-align: right;
          border-radius: 5px;
        }
    
        input:focus {
          outline: none;
          border-color: $color-picton-blue;
        }
    
        input:disabled {
          color: #000;
          border-width: 0;
        }
    
        .radios {
          flex: 45%;
          width: 45%;
          @include directionX;
          justify-content: space-around;
          gap: 1rem;
    
          .radio {
            @include directionX;
            gap: 0.25rem;
          }
        }
    
        .datePickerWrapper {
          @include directionX;
          flex: 45%;
  
          .datePicker {
            align-self: flex-end;
          }
        }
      }

      .errorRowReverse {
        width: 100%;
        margin-top: -0.5rem;
        padding: 0 0.5rem;
        @include directionX;
        justify-content: flex-end;
        color: $color-strawberry-red;
        font-size: 70%;

        .errorMsg {
          text-align: right;
        }
      }

      .errorRow {
        width: 100%;
        margin-top: -0.5rem;
        padding: 0 0.5rem;
        @include directionX;
        color: $color-strawberry-red;
        font-size: 70%;
      }
    }

    .mainError {
      margin-top: 0.5rem;
      margin-bottom: -0.5rem;
      text-align: center;
      color: $color-strawberry-red;
      font-size: 70%;
    }

    .buttons {
      margin-top: 1rem;
      @include directionY;
      gap: 1rem;
    }
  }

  @media screen and (max-width: $screen-lg-limit) {
    margin-left: 192px;
    width: calc(100vw - 192px);
  }

  @media screen and (max-width: $screen-md-limit) {
    margin-left: 10rem;
    width: calc(100vw - 10rem);
  }

  @media screen and (max-width: $screen-sm-limit) {
    margin-left: 0;
    width: 100vw;
    margin-bottom: 76px;
    height: 100%;
    min-height: calc(100vh - 76px);

    .container {
      header {
        height: 3rem;
      }
    }
  }
}