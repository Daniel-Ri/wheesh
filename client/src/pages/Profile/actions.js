import { GET_PROFILE, SET_PROFILE, UPDATE_PROFILE } from './constants';

export const setProfile = (profile) => ({
  type: SET_PROFILE,
  profile,
});

export const getProfile = () => ({
  type: GET_PROFILE,
});

export const updateProfile = (inputs, handleSuccess, handleError) => ({
  type: UPDATE_PROFILE,
  inputs,
  handleSuccess,
  handleError,
});
