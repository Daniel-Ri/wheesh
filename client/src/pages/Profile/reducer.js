import { produce } from 'immer';
import { SET_PROFILE } from './constants';

export const initialState = {
  profile: null,
};

export const storedKey = [];

const profileReducer = (state = initialState, action) =>
  produce(state, (draft) => {
    switch (action.type) {
      case SET_PROFILE:
        draft.profile = action.profile;
        break;
    }
  });

export default profileReducer;
