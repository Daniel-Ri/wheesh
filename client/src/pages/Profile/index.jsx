/* eslint-disable jsx-a11y/label-has-associated-control */
import { connect, useDispatch } from 'react-redux';
import PropTypes from 'prop-types';
import { FormattedMessage, injectIntl } from 'react-intl';
import { useNavigate } from 'react-router-dom';

import { useEffect, useState } from 'react';
import BackBtn from '@components/BackBtn';
import { createStructuredSelector } from 'reselect';
import toast from 'react-hot-toast';
import DatePicker from 'react-datepicker';
import 'react-datepicker/dist/react-datepicker.css';
import './custom.css';

import { Button } from '@mui/material';
import { selectLocale } from '@containers/App/selectors';
import { enGB, id } from 'date-fns/locale';
import classes from './style.module.scss';
import { getProfile, updateProfile } from './actions';
import { selectProfile } from './selectors';

const Profile = ({ profile, locale, intl: { formatMessage } }) => {
  const dispatch = useDispatch();
  const navigate = useNavigate();

  const [isEdit, setIsEdit] = useState(false);

  const [inputs, setInputs] = useState({
    username: '',
    gender: '',
    dateOfBirth: null,
    idCard: '',
    name: '',
    email: '',
  });

  const [errors, setErrors] = useState({
    username: '',
    dateOfBirth: '',
    idCard: '',
    name: '',
  });

  const [mainError, setMainError] = useState('');

  const today = new Date();
  const isDateAvailable = (date) => date <= today;

  const handleInputChange = (e) => {
    setInputs((prev) => ({ ...prev, [e.target.name]: e.target.value }));
  };

  const validateUsername = () => {
    if (!inputs.username) {
      setErrors((prev) => ({ ...prev, username: formatMessage({ id: 'app_username_cannot_be_empty' }) }));
      return false;
    }

    setErrors((prev) => ({ ...prev, username: '' }));
    return true;
  };

  const validateDateOfBirth = () => {
    if (!inputs.dateOfBirth) {
      setErrors((prev) => ({ ...prev, dateOfBirth: formatMessage({ id: 'app_must_choose_date_birth' }) }));
      return false;
    }

    setErrors((prev) => ({ ...prev, dateOfBirth: '' }));
    return true;
  };

  const validateIdCard = () => {
    if (!inputs.idCard) {
      setErrors((prev) => ({ ...prev, idCard: formatMessage({ id: 'app_must_insert_id_card' }) }));
      return false;
    }
    if (inputs.idCard.length !== 16 || !/^\d+$/.test(inputs.idCard)) {
      setErrors((prev) => ({ ...prev, idCard: formatMessage({ id: 'app_incorrect_id_card_format' }) }));
      return false;
    }

    setErrors((prev) => ({ ...prev, idCard: '' }));
    return true;
  };

  const validateName = () => {
    if (!inputs.name) {
      setErrors((prev) => ({ ...prev, name: formatMessage({ id: 'app_must_insert_name' }) }));
      return false;
    }

    setErrors((prev) => ({ ...prev, name: '' }));
    return true;
  };

  const handleFocusOut = (e) => {
    if (e.target.name === 'username') {
      validateUsername();
    } else if (e.target.name === 'dateOfBirth') {
      validateDateOfBirth();
    } else if (e.target.name === 'idCard') {
      validateIdCard();
    } else if (e.target.name === 'name') {
      validateName();
    }
  };

  const validateInputs = () => {
    const validatedUsername = validateUsername();
    const validatedDateOfBirth = validateDateOfBirth();
    const validatedIdCard = validateIdCard();
    const validatedName = validateName();

    if (!validatedUsername || !validatedDateOfBirth || !validatedIdCard || !validatedName) {
      return false;
    }

    return true;
  };

  const handleSuccess = () => {
    toast.success(formatMessage({ id: 'app_success_modify_profile' }));
    navigate('/me');
  };

  const handleError = (errorMsg) => {
    setMainError(errorMsg);
  };

  const handleSubmit = () => {
    setErrors({ username: '', dateOfBirth: '', idCard: '', name: '' });
    setMainError('');
    if (!validateInputs()) return;

    const formattedInputs = { ...inputs };
    formattedInputs.dateOfBirth = new Date(inputs.dateOfBirth).toDateString();
    delete formattedInputs.email;

    dispatch(updateProfile(inputs, handleSuccess, handleError));
  };

  useEffect(() => {
    dispatch(getProfile());
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  useEffect(() => {
    if (profile) {
      setInputs({
        username: profile.username,
        gender: profile.Passengers[0].gender,
        dateOfBirth: new Date(profile.Passengers[0].dateOfBirth),
        idCard: profile.Passengers[0].idCard,
        name: profile.Passengers[0].name,
        email: profile.email,
      });
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [profile]);

  return (
    <main data-testid="Profile" className={classes.main}>
      <div className={classes.container}>
        <header>
          <BackBtn handleClickBack={() => navigate('/me')} />
          <div className={classes.empty} />
          <h1>Profile</h1>
          {profile && !isEdit ? (
            <div className={classes.edit} onClick={() => setIsEdit(true)}>
              Edit
            </div>
          ) : (
            <div className={classes.empty} />
          )}
        </header>

        <form>
          <div className={classes.header}>
            <FormattedMessage id="app_personal_information" />
          </div>
          <div className={classes.input}>
            <label htmlFor="username">Username</label>
            <input
              type="text"
              name="username"
              id="username"
              value={inputs.username}
              onChange={handleInputChange}
              onBlur={handleFocusOut}
              disabled={!isEdit}
              placeholder={formatMessage({ id: 'app_enter_the_username' })}
              autoComplete="off"
            />
          </div>
          {errors.username && (
            <div className={classes.errorRowReverse}>
              <div className={classes.errorMsg}>{errors.username}</div>
            </div>
          )}

          <div className={classes.input}>
            <label htmlFor="gender">
              <FormattedMessage id="app_gender" />
            </label>
            <div className={classes.radios}>
              <div className={classes.radio} onClick={() => setInputs((prev) => ({ ...prev, gender: 'Male' }))}>
                <input
                  type="radio"
                  name="gender"
                  value="Male"
                  checked={inputs.gender === 'Male'}
                  onChange={handleInputChange}
                  disabled={!isEdit}
                />
                <label>
                  <FormattedMessage id="app_male" />
                </label>
              </div>
              <div className={classes.radio} onClick={() => setInputs((prev) => ({ ...prev, gender: 'Female' }))}>
                <input
                  type="radio"
                  name="gender"
                  value="Female"
                  checked={inputs.gender === 'Female'}
                  onChange={handleInputChange}
                  disabled={!isEdit}
                />
                <label>
                  <FormattedMessage id="app_female" />
                </label>
              </div>
            </div>
          </div>
          <div className={classes.input}>
            <label htmlFor="dateOfBirth">
              <FormattedMessage id="app_date_of_birth" />
            </label>
            <div className={classes.datePickerWrapper}>
              <DatePicker
                name="dateOfBirth"
                locale={locale === 'id' ? id : enGB}
                placeholderText={formatMessage({ id: 'app_select_your_date_of_birth' })}
                selected={inputs.dateOfBirth}
                showMonthDropdown
                dropdownMode="select"
                showYearDropdown
                dateFormat="dd/MM/yyyy"
                filterDate={isDateAvailable}
                className={classes.datePicker}
                onChange={(date) => setInputs((prev) => ({ ...prev, dateOfBirth: date }))}
                onBlur={handleFocusOut}
                disabled={!isEdit}
              />
            </div>
          </div>
          {errors.dateOfBirth && (
            <div className={classes.errorRowReverse}>
              <div className={classes.errorMsg}>{errors.dateOfBirth}</div>
            </div>
          )}

          <div className={classes.header}>
            <FormattedMessage id="app_certificate_information" />
          </div>
          <div className={classes.input}>
            <label htmlFor="idCard">
              <FormattedMessage id="app_id_card" />
            </label>
            <input
              type="text"
              name="idCard"
              id="idCard"
              value={inputs.idCard}
              onChange={handleInputChange}
              onBlur={handleFocusOut}
              disabled={!isEdit}
              placeholder={formatMessage({ id: 'app_please_enter_your_id' })}
              autoComplete="off"
            />
          </div>
          {errors.idCard && (
            <div className={classes.errorRowReverse}>
              <div className={classes.errorMsg}>{errors.idCard}</div>
            </div>
          )}

          <div className={classes.input}>
            <label htmlFor="name">
              <FormattedMessage id="app_name" />
            </label>
            <input
              type="text"
              name="name"
              id="name"
              value={inputs.name}
              onChange={handleInputChange}
              onBlur={handleFocusOut}
              disabled={!isEdit}
              placeholder={formatMessage({ id: 'app_enter_name_on_your_id' })}
              autoComplete="off"
            />
          </div>
          {errors.name && (
            <div className={classes.errorRowReverse}>
              <div className={classes.errorMsg}>{errors.name}</div>
            </div>
          )}

          <div className={classes.header}>
            <FormattedMessage id="app_contact_information" />
          </div>
          <div className={classes.input}>
            <label htmlFor="email">E-mail</label>
            <input
              type="text"
              name="email"
              id="email"
              value={inputs.email}
              onChange={handleInputChange}
              onBlur={handleFocusOut}
              disabled
              autoComplete="off"
            />
          </div>
        </form>

        {mainError && <div className={classes.mainError}>{mainError}</div>}

        {isEdit && (
          <div className={classes.buttons}>
            <Button variant="contained" className={classes.submit} onClick={handleSubmit}>
              <FormattedMessage id="app_submit_changes" />
            </Button>
          </div>
        )}
      </div>
    </main>
  );
};

Profile.propTypes = {
  profile: PropTypes.object,
  locale: PropTypes.string,
  intl: PropTypes.object,
};

const mapStateToProps = createStructuredSelector({
  profile: selectProfile,
  locale: selectLocale,
});

export default injectIntl(connect(mapStateToProps)(Profile));
