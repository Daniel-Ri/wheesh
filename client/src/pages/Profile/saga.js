import { getProfile, updateProfile } from '@domain/api';
import { takeLatest, call, put } from 'redux-saga/effects';
import { setUser } from '@containers/Client/actions';
import { setProfile } from './actions';
import { GET_PROFILE, UPDATE_PROFILE } from './constants';

function* doGetProfile() {
  try {
    const response = yield call(getProfile);
    yield put(setProfile(response.data));
  } catch (error) {
    // error
  }
}

function* doUpdateProfile({ inputs, handleSuccess, handleError }) {
  try {
    const response = yield call(updateProfile, inputs);
    yield put(setUser(response.data));
    yield call(handleSuccess);
  } catch (error) {
    yield call(handleError, error.response.data.message);
  }
}

export default function* profileSaga() {
  yield takeLatest(GET_PROFILE, doGetProfile);
  yield takeLatest(UPDATE_PROFILE, doUpdateProfile);
}
