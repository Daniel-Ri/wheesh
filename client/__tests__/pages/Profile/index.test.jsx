import { render, fireEvent } from '@utils/testHelper';
import Profile from '@pages/Profile';

const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  useNavigate: () => mockNavigate,
}));

let wrapper;

const mockProps = {
  profile: {
    id: 1,
    username: 'bangjoe',
    role: 'admin',
    email: 'bangjoe@gmail.com',
    Passengers: [
      {
        id: 1,
        userId: 1,
        isUser: true,
        gender: 'Male',
        dateOfBirth: '1995-10-13T17:00:00.000Z',
        idCard: '1243567890123456',
        name: 'Bang Joe',
        email: 'bangjoe@gmail.com',
      },
    ],
  },
};

beforeEach(() => {
  wrapper = render(<Profile {...mockProps} />);
});

describe('Page Profile', () => {
  test('Renders Profile page', () => {
    const { getByTestId } = wrapper;
    expect(getByTestId('Profile')).toBeInTheDocument();
  });

  test('Should call navigate when BackBtn clicked', () => {
    const { getByTestId } = wrapper;
    const backButton = getByTestId('BackBtn');
    fireEvent.click(backButton);
    expect(mockNavigate).toHaveBeenCalledWith('/me');
  });

  test('Should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
