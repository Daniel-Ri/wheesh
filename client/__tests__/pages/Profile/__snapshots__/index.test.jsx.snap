// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Profile Should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="main"
        data-testid="Profile"
      >
        <div
          class="container"
        >
          <header>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium btn css-1e6y48t-MuiButtonBase-root-MuiButton-root"
              data-testid="BackBtn"
              tabindex="0"
              type="button"
            >
              Kembali
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <div
              class="empty"
            />
            <h1>
              Profile
            </h1>
            <div
              class="empty"
            />
          </header>
          <form>
            <div
              class="header"
            >
              Informasi Pribadi
            </div>
            <div
              class="input"
            >
              <label
                for="username"
              >
                Username
              </label>
              <input
                autocomplete="off"
                disabled=""
                id="username"
                name="username"
                placeholder="Masukkan username"
                type="text"
                value=""
              />
            </div>
            <div
              class="input"
            >
              <label
                for="gender"
              >
                Jenis Kelamin
              </label>
              <div
                class="radios"
              >
                <div
                  class="radio"
                >
                  <input
                    disabled=""
                    name="gender"
                    type="radio"
                    value="Male"
                  />
                  <label>
                    Pria
                  </label>
                </div>
                <div
                  class="radio"
                >
                  <input
                    disabled=""
                    name="gender"
                    type="radio"
                    value="Female"
                  />
                  <label>
                    Wanita
                  </label>
                </div>
              </div>
            </div>
            <div
              class="input"
            >
              <label
                for="dateOfBirth"
              >
                Tanggal Kelahiran
              </label>
              <div
                class="datePickerWrapper"
              >
                <div
                  class="react-datepicker-wrapper"
                >
                  <div
                    class="react-datepicker__input-container"
                  >
                    <input
                      class="datePicker"
                      disabled=""
                      name="dateOfBirth"
                      placeholder="Pilih tanggal lahir"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="header"
            >
              Informasi Sertifikat
            </div>
            <div
              class="input"
            >
              <label
                for="idCard"
              >
                Kartu Identitas
              </label>
              <input
                autocomplete="off"
                disabled=""
                id="idCard"
                name="idCard"
                placeholder="Silakan masukkan KTP Indonesia Anda"
                type="text"
                value=""
              />
            </div>
            <div
              class="input"
            >
              <label
                for="name"
              >
                Nama
              </label>
              <input
                autocomplete="off"
                disabled=""
                id="name"
                name="name"
                placeholder="Masukkan nama Anda sesuai KTP"
                type="text"
                value=""
              />
            </div>
            <div
              class="header"
            >
              Informasi Kontak
            </div>
            <div
              class="input"
            >
              <label
                for="email"
              >
                E-mail
              </label>
              <input
                autocomplete="off"
                disabled=""
                id="email"
                name="email"
                type="text"
                value=""
              />
            </div>
          </form>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="main"
      data-testid="Profile"
    >
      <div
        class="container"
      >
        <header>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium btn css-1e6y48t-MuiButtonBase-root-MuiButton-root"
            data-testid="BackBtn"
            tabindex="0"
            type="button"
          >
            Kembali
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <div
            class="empty"
          />
          <h1>
            Profile
          </h1>
          <div
            class="empty"
          />
        </header>
        <form>
          <div
            class="header"
          >
            Informasi Pribadi
          </div>
          <div
            class="input"
          >
            <label
              for="username"
            >
              Username
            </label>
            <input
              autocomplete="off"
              disabled=""
              id="username"
              name="username"
              placeholder="Masukkan username"
              type="text"
              value=""
            />
          </div>
          <div
            class="input"
          >
            <label
              for="gender"
            >
              Jenis Kelamin
            </label>
            <div
              class="radios"
            >
              <div
                class="radio"
              >
                <input
                  disabled=""
                  name="gender"
                  type="radio"
                  value="Male"
                />
                <label>
                  Pria
                </label>
              </div>
              <div
                class="radio"
              >
                <input
                  disabled=""
                  name="gender"
                  type="radio"
                  value="Female"
                />
                <label>
                  Wanita
                </label>
              </div>
            </div>
          </div>
          <div
            class="input"
          >
            <label
              for="dateOfBirth"
            >
              Tanggal Kelahiran
            </label>
            <div
              class="datePickerWrapper"
            >
              <div
                class="react-datepicker-wrapper"
              >
                <div
                  class="react-datepicker__input-container"
                >
                  <input
                    class="datePicker"
                    disabled=""
                    name="dateOfBirth"
                    placeholder="Pilih tanggal lahir"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="header"
          >
            Informasi Sertifikat
          </div>
          <div
            class="input"
          >
            <label
              for="idCard"
            >
              Kartu Identitas
            </label>
            <input
              autocomplete="off"
              disabled=""
              id="idCard"
              name="idCard"
              placeholder="Silakan masukkan KTP Indonesia Anda"
              type="text"
              value=""
            />
          </div>
          <div
            class="input"
          >
            <label
              for="name"
            >
              Nama
            </label>
            <input
              autocomplete="off"
              disabled=""
              id="name"
              name="name"
              placeholder="Masukkan nama Anda sesuai KTP"
              type="text"
              value=""
            />
          </div>
          <div
            class="header"
          >
            Informasi Kontak
          </div>
          <div
            class="input"
          >
            <label
              for="email"
            >
              E-mail
            </label>
            <input
              autocomplete="off"
              disabled=""
              id="email"
              name="email"
              type="text"
              value=""
            />
          </div>
        </form>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
